### ISSUE TITLE
loop when sequence is not closed

### ISSUE THREAD

**Issue by miekg** [miekg/cf#44]
this causes a loop, the semicolon is not found and the closing brace isn't eaten
~~~
body common control
{
      bundlesequence => {
                          @(update_def.bundlesequence_end),
                         "hi",
      }
}
~~~

**Comment by miekg** [comment]
this is because of this:
~~~ go
func BodySelections(b *rd.Builder) (ok bool) {
	b.Enter("BodySelections")
	defer b.Exit(&ok)

	for {
		Comments(b) // comments in between selections and trailing ones
		if !Selection(b) {
			return true
		}
		Macro(b)
	}
}
~~~
returning true makes us return here again, for `bundle` I apparently correctly fixed it...

so not, it has the same code in 'Promises', but haven't triggered that with bogus cf code

**Comment by miekg** [comment]
ugly fix/hack in #45
